rns_shell/platform/graphics/x11/WindowX11.cpp

```diff
@@ -58,13 +63,38 @@ void Window::createEventLoop(Application* app) {
             while (count-- && !done) {
                 XEvent event;
                 XNextEvent(display, &event);
-
                 switch (event.type) {
-                    case ConfigureNotify:
-                        RNS_LOG_INFO("Resize Request with (Width x Height) : (" << event.xconfigurerequest.width <<
+                    case ConfigureNotify:{
+
+                        if(event.xconfigurerequest.window == DefaultRootWindow(display)) {
+                            RNS_LOG_INFO(" ROOT Window(Screen) Size :" << event.xconfigurerequest.width << "x" << event.xconfigurerequest.height);
+
+                            if(((RnsShell::PlatformDisplay::sharedDisplay().getCurrentScreenSize().width()!=event.xconfigurerequest.width) ||
+                                (PlatformDisplay::sharedDisplay().getCurrentScreenSize().height()!=event.xconfigurerequest.height)) &&
+                                ((PlatformDisplay::sharedDisplay().getCurrentScreenSize().height()!=-1) ||
+                                (PlatformDisplay::sharedDisplay().getCurrentScreenSize().width()!=-1))) {
+
+                                RnsShell::PlatformDisplay::sharedDisplay().setCurrentScreenSize(event.xconfigurerequest.width,event.xconfigurerequest.height);
```

Comment:
setCurrentScreenSize is common in both if and else condition. So can it be brought outside. 
If yes, then else condition is not required.

Reply:
No, else condition is required.Default currnentScreenSize value is -1. To avoid the first triggering event when comparing currentScreenSize with actual/original screen size need to assign/set a new value that means actual/original screen size.

Reply:
Why are we doing -1 check ? Is it done to avoid any additional event ? Ideally in constructor we have to assign valid value and keep.